src/D14/Emulator.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace aoc.D14
+{
+  public class Emulator
+  {
+    private Dictionary<long, String> _memory;
+
+    public Emulator()
+    {
+      _memory = new Dictionary<long, string>();
+    }
+
+    public long PartI(List<Tuple<string, long, long>> list) 
+    {
+      string mask;
+      long address;
+      long value;
+      long sum = 0;
+
+      foreach (var item in list)
+      {
+        mask = item.Item1;
```

Comment:
@taherkheli I think this can also be written as `var (mask, address, value) = item;

Reply:
or you can put the destructoring in the foreach declaration

Reply:
see more [here](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/value-tuples)

Reply:
all great but you should have written `(mask, address, value) = item;` instead ;) .. fixed